tracer/test/Datadog.Trace.Tests/ExtensionMethods/SpanExtensionsTests.cs

```diff
@@ -34,26 +34,14 @@ public SpanExtensionsTests()
             var spanContext = new SpanContext(Mock.Of<ISpanContext>(), Mock.Of<ITraceContext>(), "test");
             var span = new Span(spanContext, null);
 
-            span.AddTagsFromDbCommand(CreateDbCommand(connectionString));
+            var commandTags = DbCommandCache.GetTagsFromDbCommand(CreateDbCommand(connectionString));
 
-            Assert.Equal(expectedDbName, span.GetTag(Tags.DbName));
-            Assert.Equal(expectedUserId, span.GetTag(Tags.DbUser));
-            Assert.Equal(expectedHost, span.GetTag(Tags.OutHost));
-        }
-
-        [Fact]
-        public void SetSpanTypeToSql()
-        {
-            const string connectionString = "Server=myServerName;Database=myDataBase;User Id=myUsername;Password=myPassword;";
-            const string commandText = "SELECT * FROM Table ORDER BY id";
-
-            var spanContext = new SpanContext(Mock.Of<ISpanContext>(), Mock.Of<ITraceContext>(), "test");
-            var span = new Span(spanContext, null);
-
-            span.AddTagsFromDbCommand(CreateDbCommand(connectionString, commandText));
-
-            Assert.Equal(SpanTypes.Sql, span.Type);
-            Assert.Equal(commandText, span.ResourceName);
+            if (commandTags != null)
+            {
+                Assert.Equal(expectedDbName, commandTags.Value.DbName);
+                Assert.Equal(expectedUserId, commandTags.Value.DbUser);
+                Assert.Equal(expectedHost, commandTags.Value.OutHost);
+            }
```

Comment:
commandTags shouldn't be null here right? though we should probably test a case where it is expected to be?

```suggestion
            Assert.NotNull(commandTags);
            Assert.Equal(expectedDbName, commandTags.Value.DbName);
            Assert.Equal(expectedUserId, commandTags.Value.DbUser);
            Assert.Equal(expectedHost, commandTags.Value.OutHost);
```

Reply:
Right, I remove the Nullable<T> in 769ed59